### PR TITLE
feat(recovery keys): added upsertRecoveryKey db method

### LINKED ISSUE #4268: Consider using using upsertRecoveryKey database method
From https://github.com/mozilla/fxa/pull/4258/files/672e8c4e618646925678de6c2114c1d22f0d7726#diff-a4f2c3f852c9bcb0ac86a24d33a38b39

It might be a little bit cleaner if this was abstracted to the database and let it do the necessary cleanup for unconfirmed recovery keys.

### PR DESCRIPTION
**Pull request by farhan787**
fixes #4268

### CODE REVIEW

**Comment by farhan787** [comment]
@vbudhram I've updated this PR as the PR responsible for updating procedures for setting`verifiedAt` column in `email` table has been merged. IMHO I think it's ready for your final review, please take a look at it.

**Comment by farhan787** [comment]
@vbudhram [this test (should fail to create multiple keys)](https://github.com/mozilla/fxa/blob/9a52a7efdbe5bd447b0b10d6ce4ab6636e8de899/packages/fxa-auth-server/test/remote/recovery_key_tests.js#L209) is failing in `auth-server` because the `recoveryKey` isn't verified there and as we've replaced the `createRecoveryKey` method with `upsertRecoveryKey` method which creates a new unverified recovery key if a verified recoveryKey doesn't exist. 

To fix this we can do 2 things, 
- Either verify the recoveryKey in `auth-server` test which I think might not be right as we'd be using upsertRecoveryKey in `auth-db-mysql` and treating it as `createRecoveryKey` in `auth-server`. 
- Or have both `createRecoveryKey` and `upsertRecoveryKey` methods for client to call.

Waiting for your instructions.

File: packages/fxa-auth-db-mysql/lib/db/schema/patch-109-110.sql [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  -- Signal error if no user found
+  SET @accountCount = 0;
+  SELECT COUNT(*) INTO @accountCount FROM `accounts` WHERE `uid` = `uidArg`;
+  IF @accountCount = 0 THEN
+    SIGNAL SQLSTATE '45000' SET MYSQL_ERRNO = 1643, MESSAGE_TEXT = 'Can not create recovery key for unknown user.';
+  END IF;
+
+  -- Signal error if verified recovery key already exists
+  SET @verifiedRecoveryKeyCount = 0;
+  SELECT COUNT(*) INTO @verifiedRecoveryKeyCount FROM `recoveryKeys` WHERE `uid` = `uidArg` AND verifiedAt IS NOT NULL;
```
Comment (vbudhram) [inline packages/fxa-auth-db-mysql/lib/db/schema/patch-109-110.sql:37]: I've double checked with our OPs team and it should be good to use select on the `enabled` column verses `verifiedAt`.

**Comment by farhan787** [comment]
Blocked because of this, as my local dev environment is broken https://github.com/mozilla/fxa/issues/5216

**Comment by clouserw** [comment]
Please rebase/reopen when ready for merging

**Pull request closed without merging**